src/libraries/System.Collections/tests/Generic/PriorityQueue/PriorityQueue.Generic.Tests.cs

```diff
@@ -0,0 +1,291 @@
+            Assert.Empty(queue.UnorderedItems);
+            Assert.Equal(queue.Comparer, Comparer<TPriority>.Default);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Constructor_int(int initialCapacity)
+        {
+            var queue = new PriorityQueue<TElement, TPriority>(initialCapacity);
+
+            Assert.Empty(queue.UnorderedItems);
+        }
+
+        [Fact]
+        public void PriorityQueue_Generic_Constructor_int_Negative_ThrowsArgumentOutOfRangeException()
+        {
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("initialCapacity", () => new PriorityQueue<TElement, TPriority>(-1));
+            AssertExtensions.Throws<ArgumentOutOfRangeException>("initialCapacity", () => new PriorityQueue<TElement, TPriority>(int.MinValue));
+        }
+
+        [Fact]
+        public void PriorityQueue_Generic_Constructor_IComparer()
+        {
+            IComparer<TPriority> comparer = Comparer<TPriority>.Default;
+            var queue = new PriorityQueue<TElement, TPriority>(comparer);
+
+            Assert.Equal(comparer, queue.Comparer);
+        }
+
+        [Fact]
+        public void PriorityQueue_Generic_Constructor_IComparer_Null()
+        {
+            var queue = new PriorityQueue<TElement, TPriority>((IComparer<TPriority>)null);
+            Assert.Equal(Comparer<TPriority>.Default, queue.Comparer);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Constructor_int_IComparer(int initialCapacity)
+        {
+            IComparer<TPriority> comparer = Comparer<TPriority>.Default;
+            var queue = new PriorityQueue<TElement, TPriority>(initialCapacity);
+
+            Assert.Empty(queue.UnorderedItems);
+            Assert.Equal(comparer, queue.Comparer);
+        }
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Constructor_IEnumerable(int count)
+        {
+            HashSet<(TElement, TPriority)> itemsToEnqueue = this.GenericIEnumerableFactory(count).ToHashSet();
+            PriorityQueue<TElement, TPriority> queue = new PriorityQueue<TElement, TPriority>(itemsToEnqueue);
+            Assert.True(itemsToEnqueue.SetEquals(queue.UnorderedItems));
+        }
+
+        #endregion
+
+        #region Enqueue, Dequeue, Peek
+
+        [Theory]
+        [MemberData(nameof(ValidCollectionSizes))]
+        public void PriorityQueue_Generic_Enqueue(int count)
+        {
+            PriorityQueue<TElement, TPriority> queue = GenericPriorityQueueFactory(count, count, out var generatedItems);
+            HashSet<(TElement, TPriority)> expectedItems = generatedItems.ToHashSet();
+
+            Assert.Equal(count, queue.Count);
+            var actualItems = queue.UnorderedItems.ToArray();
+            foreach (var (element, priority) in actualItems)
```

Comment:
This doesn't verify that `queue.UnorderedItems` returned anything at all.

I can't find anything like `Assert.SetEqual(IEnumerable<T>, IEnumerable<T>, IEqualityComparer<T> = null)` strangely.

Instad this should work `Assert.Equal(new HashSet<(TElement, TPriority)> generatedItems, new HashSet<(TElement, TPriority)>queue.UnorderedItems.ToArray()))`

Reply:
Although, I think you probably want to validate enumeration ordering. Although it's "UnorderedItems" what I think it means is "consistently ordered items until you modify the collectoin" and possibly (I didn't check) "consistently ordered items for any particular collectoin content"
Either way, we are implicitly promising the order doesn't change if enumerate twice - we should therefore test that.

Reply:
Went with

```
var actualItems = queue.UnorderedItems.ToArray();
Assert.True(expectedItems.SetEquals(actualItems));
```